File: Source/ServiceRegister.UserManagement/Mapping/UserListItemMapper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Affecto.Mapping;
using Affecto.IdentityManagement.Interfaces.Model;
using ServiceRegister.UserManagement.Model;

namespace ServiceRegister.UserManagement.Mapping
{
    internal class UserListItemMapper : IMapper<IUser, UserListItem>
    {
        private readonly IMapper<IRole, Role> roleMapper;

        public UserListItemMapper(IMapper<IRole, Role> roleMapper)
        {
            if (roleMapper == null)
            {
                throw new ArgumentNullException("roleMapper");
            }
            this.roleMapper = roleMapper;
        }

        public UserListItem Map(IUser source)
        {
            if (source == null)
            {
                return null;
            }

            if (source.CustomProperties == null)
            {
                throw new ArgumentException("User custom properties cannot be null.", "source");
            }

            List<KeyValuePair<string, string>> keyValuePairs = source.CustomProperties
                .Select(c => new KeyValuePair<string, string>(c.Name, c.Value))
                .ToList();

            var customProperties = new CustomProperties(keyValuePairs);
            customProperties.ValidateRequiredProperties();

            var result = new UserListItem
            {
                Id = source.Id,
                Role = GetRole(source),
                OrganizationId = customProperties.OrganizationId.Value,
                EmailAddress = customProperties.EmailAddress,
                LastName = customProperties.LastName,
                FirstName = customProperties.FirstName
            };

            return result;
        }

        private Role GetRole(IUser user)
        {
            if (user.Roles == null)
            {
                throw new ArgumentException("User roles cannot be null.", "user");
            }
            if (user.Roles.Count != 1)
            {
                throw new ArgumentException("User must have a single role.", "user");
            }

            return roleMapper.Map(user.Roles.Single());
        }
    }
}

File: Source/AngularApplication/App/Mapping/UserListItemMapper.ts
"use strict";

module ServiceRegister
{
    export class UserListItemMapper
    {
        public static map(data: any): Array<UserListItem>
        {
            var result: Array<UserListItem> = new Array<UserListItem>();
            data.forEach((item: any) =>
            {
                result.push(this.mapSingle(item));
            });
            return result;
        }

        public static mapSingle(data: any): UserListItem
        {
            var organization: OrganizationName = OrganizationNameMapper.mapSingle(data.organization);
            var role: UserRole = UserRoleMapper.mapSingle(data.role);
            return new UserListItem(data.id, role, organization, data.emailAddress, data.lastName, data.firstName);
        }
    }
}